ui/src/main/java/edu/wpi/grip/ui/pipeline/SocketHandleView.java

```diff
@@ -35,22 +33,35 @@
     private static final PseudoClass CONNECTING_PSEUDO_CLASS = PseudoClass.getPseudoClass("connecting");
     private static final PseudoClass CONNECTED_PSEUDO_CLASS = PseudoClass.getPseudoClass("connected");
 
-    private static Optional<Socket> draggingSocket = Optional.empty();
+    private final EventBus eventBus;
+    private final Socket socket;
 
-    final private Socket socket;
-    final private EventBus eventBus;
+    /**
+     * Provides a singleton object to assign the socket being dragged from during dragging to allow for a
+     * connection to be made.
+     */
+    @Singleton
+    protected static final class SocketHandleDraggingSocketService {
+        private Optional<Socket> draggingSocket = Optional.empty();
+    }
 
     final private BooleanProperty connectingProperty = new SimpleBooleanProperty(this, "connecting", false);
     final private BooleanProperty connectedProperty = new SimpleBooleanProperty(this, "connected", false);
 
-    public SocketHandleView(EventBus eventBus, Socket socket) {
-        checkNotNull(eventBus);
-        checkNotNull(socket);
-
-        this.setTooltip(new Tooltip("Drag to connect"));
+    public interface Factory {
+        SocketHandleView create(Socket socket);
+    }
 
-        this.socket = socket;
+    @Inject
+    SocketHandleView(EventBus eventBus,
```

Comment:
Since most of the combined view/controller classes are being turned into controllers and this one isn't, drop the *View naming and just call this `SocketHandle`.  This is more consistent with every other javafx Node subclass (the *View thing was just to avoid name conflicts between things like `Pipeline` and `PipelineView`)